code/segmentedtickerplant/stplog.q

```diff
@@ -290,14 +290,24 @@ init:{[dbname]
     // add the info to the meta table
     .stpm.updmeta[multilog][`open;logtabs;.z.p+.eodtime.dailyadj];
     ]
+
+  // set log handles to null in sctp if not in create mode
+  if[.sctp.chainedtp and not .sctp.loggingmode=`create; 
+    `..loghandles set .stplg.t!(count .stplg.t)#(::);
+    // only need to log errors if sctp is creating its own logs
+    `.stplg.badmsg set {[x;y;z]};
+    ];
  };
 
 \d .
 
 // Close logs on clean exit
 .z.exit:{
   if[not x~0i;.lg.e[`stpexit;"Bad exit!"];:()];
-  .lg.o[`stpexit;"Exiting process and closing off log files."];
-  .stpm.updmeta[.stplg.multilog][`close;.stpps.t;.z.p];
-  .stplg.closelog each .stpps.t;
+  .lg.o[`stpexit;"Exiting process"];
+  if[.sctp.loggingmode=`create; // only close logs if sctp is in create mode
```

Comment:
This still needs to be triggered for normal STP usage, probably do something like if its an SCTP and mode is not create then `:()`

Reply:
ah i see what you mean, yeah ill just stop the execution early instead